crates/node_binding/src/plugins/interceptor.rs

```diff
@@ -660,6 +679,28 @@ impl AsyncSeries<Compilation> for CompilationFinishModulesTap {
   }
 }
 
+#[async_trait]
+impl AsyncSeriesBail<Compilation, bool> for CompilationOptimizeModulesTap {
+  async fn run(&self, _compilation: &mut Compilation) -> rspack_error::Result<Option<bool>> {
+    self.function.call(()).await
```

Comment:
Does this allow `Promise<T>`? Otherwise, we can use `call_with_sync` to ensure the final result and avoid the overhead for determine the branch of `Either<T, Promise<T>>`

Reply:
fixed in https://github.com/web-infra-dev/rspack/pull/6003/commits/910d87383327eef6656d34e90cc413eb9846c2a0.. too many branches..